lib/class-wp-rest-plugins-controller.php

```diff
@@ -0,0 +1,829 @@
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		if ( ! current_user_can( 'delete_plugins' ) ) {
+			return new WP_Error(
+				'rest_cannot_manage_plugins',
+				__( 'Sorry, you are not allowed to delete plugins for this site.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		$can_read = $this->check_read_permission( $request['plugin'] );
+
+		if ( is_wp_error( $can_read ) ) {
+			return $can_read;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Deletes one plugin from the site.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		$data = $this->get_plugin_data( $request['plugin'] );
+
+		if ( is_wp_error( $data ) ) {
+			return $data;
+		}
+
+		if ( is_plugin_active( $request['plugin'] ) ) {
+			return new WP_Error(
+				'rest_cannot_delete_active_plugin',
+				__( 'Cannot delete an active plugin. Please deactivate it first.', 'gutenberg' ),
+				array( 'status' => 400 )
+			);
+		}
+
+		$filesystem_available = $this->is_filesystem_available();
+		if ( is_wp_error( $filesystem_available ) ) {
+			return $filesystem_available;
+		}
+
+		$deleted = delete_plugins( array( $request['plugin'] ) );
+
+		if ( is_wp_error( $deleted ) ) {
+			return $deleted;
+		}
+
+		return new WP_REST_Response( null, 204 );
+	}
+
+	/**
+	 * Prepares the plugin for the REST response.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param mixed           $item    Unmarked up and untranslated plugin data from {@see get_plugin_data()}.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$item   = _get_plugin_data_markup_translate( $item['_file'], $item, false );
+		$marked = _get_plugin_data_markup_translate( $item['_file'], $item, true, false );
+
+		$data = array(
+			'plugin'       => $item['_file'],
+			'status'       => $this->get_plugin_status( $item['_file'] ),
+			'name'         => $item['Name'],
+			'plugin_uri'   => $item['PluginURI'],
+			'author'       => $item['Author'],
+			'author_uri'   => $item['AuthorURI'],
+			'description'  => array(
+				'raw'      => $item['Description'],
+				'rendered' => $marked['Description'],
+			),
+			'version'      => $item['Version'],
+			'network_only' => $item['Network'],
+			'requires_wp'  => $item['RequiresWP'],
+			'requires_php' => $item['RequiresPHP'],
+			'text_domain'  => $item['TextDomain'],
+		);
+
+		$data = $this->add_additional_fields_to_object( $data, $request );
+
+		$response = new WP_REST_Response( $data );
```

Comment:
Lets break this add link to it's own method `protected function prepare_links`

Reply:
Good point, fixed in 3737daf82e.